### PR TITLE
Gh4994 Improve logic and add new columns to partner and user importers

### PR DESCRIPTION
**Pull request by metas-rc**
https://github.com/metasfresh/metasfresh/issues/4994

### CODE REVIEW

File: de.metas.business/src/main/java/de/metas/globalid/impexp/BPartnerGlobalIDImportProcess.java [hunk 1]
```diff
@@ -76,7 +76,9 @@ protected I_I_BPartner_GlobalID retrieveImportRecord(Properties ctx, ResultSet r
 	}
```
Comment (metas-ts) [inline de.metas.business/src/main/java/de/metas/globalid/impexp/BPartnerGlobalIDImportProcess.java]:
```suggestion
/**
  * @param isInsertOnly ignored. This import is only for updates.
  */
```

File: de.metas.business/src/main/java/de/metas/globalid/impexp/BPartnerGlobalIDImportProcess.java [hunk 2]
```diff
@@ -76,7 +76,9 @@ protected I_I_BPartner_GlobalID retrieveImportRecord(Properties ctx, ResultSet r
...
 
 	@Override
-	protected ImportRecordResult importRecord(IMutable<Object> state, I_I_BPartner_GlobalID importRecord)
+	protected ImportRecordResult importRecord(IMutable<Object> state,
+			I_I_BPartner_GlobalID importRecord,
+			final boolean isInsertOnly /* not used. This import is only for updates */ )
```
Comment (metas-ts) [inline de.metas.business/src/main/java/de/metas/globalid/impexp/BPartnerGlobalIDImportProcess.java]:
```suggestion
			final boolean isInsertOnly)
```

File: de.metas.business/src/main/java/org/adempiere/impexp/ADUserImportProcess.java [hunk 3]
```diff
@@ -157,7 +154,9 @@ protected I_I_User retrieveImportRecord(final Properties ctx, final ResultSet rs
 	}
 
 	@Override
-	protected ImportRecordResult importRecord(final IMutable<Object> state, final I_I_User importRecord) throws Exception
+	protected ImportRecordResult importRecord(final IMutable<Object> state,
+			final I_I_User importRecord,
+			final boolean isInsertOnly /* not used, this importer always inserts */ ) throws Exception
```
Comment (metas-ts) [inline de.metas.business/src/main/java/org/adempiere/impexp/ADUserImportProcess.java]: same as further up: a javadoc would be better than this comment

File: de.metas.business/src/main/java/org/adempiere/impexp/RequestImportProcess.java [hunk 4]
```diff
@@ -180,7 +180,9 @@ protected I_I_Request retrieveImportRecord(final Properties ctx, final ResultSet
 	}
 
 	@Override
-	protected ImportRecordResult importRecord(final IMutable<Object> state, final I_I_Request importRecord) throws Exception
+	protected ImportRecordResult importRecord(final IMutable<Object> state,
+			final I_I_Request importRecord,
+			final boolean isInsertOnly /* not used, this importer always inserts */) throws Exception
```
Comment (metas-ts) [inline de.metas.business/src/main/java/org/adempiere/impexp/RequestImportProcess.java]: same as further up: a javadoc would be better than this comment

File: de.metas.business/src/test/java/de/metas/bpartner/impexp/BPartnerImportProcess_MultiLocations_gh2543_Test.java [hunk 5]
```diff
@@ -76,7 +76,7 @@ public void testMultiplePartnerAndAddresses()
 
 		final IMutable<Object> state = new Mutable<>();
 
-		ibpartners.forEach(importRecord -> importProcess.importRecord(state, importRecord));
+		ibpartners.forEach(importRecord -> importProcess.importRecord(state, importRecord, false));
```
Comment (metas-ts) [inline de.metas.business/src/test/java/de/metas/bpartner/impexp/BPartnerImportProcess_MultiLocations_gh2543_Test.java]:
```suggestion
		ibpartners.forEach(importRecord -> importProcess.importRecord(state, importRecord, false/*isInsertOnly*/));
```

File: de.metas.business/src/test/java/de/metas/bpartner/impexp/BPartnerImportProcess_SimpleCase_Test.java [hunk 6]
```diff
@@ -56,7 +56,7 @@ public void testSingleBpartner()
 
 		final BPartnerImportProcess importProcess = new BPartnerImportProcess();
 		importProcess.setCtx(ctx);
-		importProcess.importRecord(new Mutable<>(), ibpartner);
+		importProcess.importRecord(new Mutable<>(), ibpartner, false);
```
Comment (metas-ts) [inline de.metas.business/src/test/java/de/metas/bpartner/impexp/BPartnerImportProcess_SimpleCase_Test.java]:
```suggestion
		importProcess.importRecord(new Mutable<>(), ibpartner, false/*isInsertOnly*/);
```

File: de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java [hunk 7]
```diff
@@ -142,7 +142,7 @@ public void testImportActiveFlatrateTerms()
 
 		final FlatrateTermImportProcess importProcess = new FlatrateTermImportProcess();
 		importProcess.setCtx(helper.getCtx());
-		importProcess.importRecord(new Mutable<>(), iflatrateTerm);
+		importProcess.importRecord(new Mutable<>(), iflatrateTerm, true);
```
Comment (metas-ts) [inline de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java]:
```suggestion
		importProcess.importRecord(new Mutable<>(), iflatrateTerm, true/*isInsertOnly*/);
```

File: de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java [hunk 8]
```diff
@@ -211,7 +211,7 @@ public void testImportActiveFlatrateTermsWithTaxIncluded()
...
```
Comment (metas-ts) [inline de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java]:
```suggestion
		importProcess.importRecord(new Mutable<>(), iflatrateTerm, true/*isInsertOnly*/);
```

File: de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java [hunk 9]
```diff
@@ -276,7 +276,7 @@ public void testImportingInActiveFlatrateTerms()
...
```
Comment (metas-ts) [inline de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java]:
```suggestion
		importProcess.importRecord(new Mutable<>(), iflatrateTerm, true/*isInsertOnly*/);
```

File: de.metas.acct.base/src/test/java/de/metas/acct/impexp/AccountImportProcess_Test.java [hunk 10]
```diff
@@ -53,7 +53,7 @@ public void testAccountImport()
 		final AccountImportProcess importProcess = new AccountImportProcess();
 		importProcess.setCtx(ctx);
 
-		ievs.forEach(iElelemntValue -> importProcess.importRecord(new Mutable<>(), iElelemntValue));
+		ievs.forEach(iElelemntValue -> importProcess.importRecord(new Mutable<>(), iElelemntValue, false));
```
[the remark this answers is not in the document]
Comment (metas-rc) [reply de.metas.acct.base/src/test/java/de/metas/acct/impexp/AccountImportProcess_Test.java]: done

File: de.metas.adempiere.adempiere/base/src/main/java-gen/org/compiere/model/I_AD_ImpFormat.java [hunk 11]
```diff
@@ -223,6 +223,29 @@
...
+	/**
+	 * Get isManualImport.
+	 *
+	 * <br>Type: YesNo
+	 * <br>Mandatory: true
+	 * <br>Virtual Column: false
+	 */
+	public boolean isManualImport();
+
+    /** Column definition for isManualImport */
```
[the remark this answers is not in the document]
Comment (metas-rc) [reply de.metas.adempiere.adempiere/base/src/main/java-gen/org/compiere/model/I_AD_ImpFormat.java]: done

File: de.metas.business/src/main/java/de/metas/globalid/impexp/BPartnerGlobalIDImportProcess.java (hunk 2 above)
Comment (metas-rc) [reply de.metas.business/src/main/java/de/metas/globalid/impexp/BPartnerGlobalIDImportProcess.java]: done

File: de.metas.business/src/main/java/org/adempiere/impexp/ADUserImportProcess.java (hunk 3 above)
Comment (metas-rc) [reply de.metas.business/src/main/java/org/adempiere/impexp/ADUserImportProcess.java]: done

File: de.metas.business/src/test/java/de/metas/bpartner/impexp/BPartnerImportProcess_MultiLocations_gh2543_Test.java (hunk 5 above)
Comment (metas-rc) [reply de.metas.business/src/test/java/de/metas/bpartner/impexp/BPartnerImportProcess_MultiLocations_gh2543_Test.java]: done

File: de.metas.business/src/test/java/de/metas/bpartner/impexp/BPartnerImportProcess_SimpleCase_Test.java (hunk 6 above)
Comment (metas-rc) [reply de.metas.business/src/test/java/de/metas/bpartner/impexp/BPartnerImportProcess_SimpleCase_Test.java]: done

File: de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java (hunk 7 above)
Comment (metas-rc) [reply de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java]: done

File: de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java (hunk 8 above)
Comment (metas-rc) [reply de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java]: done

File: de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java (hunk 9 above)
Comment (metas-rc) [reply de.metas.contracts/src/test/java/de/metas/contracts/flatrate/impexp/FlatrateTermImportProcess_SimpleCase_Test.java]: done

File: de.metas.business/src/main/java/de/metas/globalid/impexp/BPartnerGlobalIDImportProcess.java (hunk 1 above)
Comment (metas-rc) [reply de.metas.business/src/main/java/de/metas/globalid/impexp/BPartnerGlobalIDImportProcess.java]: done

File: de.metas.business/src/main/java/org/adempiere/impexp/RequestImportProcess.java (hunk 4 above)
Comment (metas-ts) [reply de.metas.business/src/main/java/org/adempiere/impexp/RequestImportProcess.java]: done

**Pull request merged**